include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -7176,11 +7176,24 @@ namespace VectorTools
               Assert (n_components >= dim,
                       ExcMessage ("You can only ask for the Hdiv norm for a finite element "
                                   "with at least 'dim' components. In that case, this function "
-                                  "will take the divergence of the first 'dim' components."));
+                                  "will find the index of the first non-zero weight and take "
+                                  "the divergence of the 'dim' components that follow it."));
+
+              unsigned int idx = 0;
+              if (weight!=nullptr)
+                {
+                  Vector<double> tmp(n_components);
+                  weight->vector_value(Point<spacedim>(), tmp);
+
+                  for (; idx<n_components; ++idx)
+                    if (fabs(tmp[idx]) > 1.e-12)
```

Comment:
I think this is overly cautious: since we suggest using `ComponentSelectFunction` we should just check that this value is positive without a tolerance.